### PR TITLE
Add VSCode task for debugging C++ in Workbench

### PR DESCRIPTION
**Pull request by dtasev**
**Description of work.**
Adds a debug task to VSCode for starting the Workbench and attaching `gdb`

**To test:**
Add task to `launch.json`. Change paths for your setup. Run task, it should start the workbench and attach `gdb`.

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: dev-docs/source/VSCode.rst [hunk 1]
```diff
@@ -100,6 +100,8 @@ attempt to make a useful task for you. However it may be better to use this exam
 The commands can be switched out with the command and various args for the generator
 used to generate your CMake with.
 
+
+
```
Comment (Pasarus) [inline dev-docs/source/VSCode.rst]:
```suggestion

```

File: dev-docs/source/VSCode.rst [hunk 2]
```diff
@@ -100,6 +100,8 @@ attempt to make a useful task for you. However it may be better to use this exam
 The commands can be switched out with the command and various args for the generator
 used to generate your CMake with.
 
+
```
Comment (Pasarus) [inline dev-docs/source/VSCode.rst]:
```suggestion

```

File: dev-docs/source/VSCode.rst [hunk 3]
```diff
@@ -186,7 +190,36 @@ you will want to make your file look something a little like this:
...
+      "MIMode": "gdb",
+      "preLaunchTask": "Build Mantid",
+      "setupCommands": [
+        {
+          "description": "Enable pretty-printing for gdb",
+          "text": "-enable-pretty-printing",
+          "ignoreFailures": true
+        }
+      ]
+    },
```
Comment (Pasarus) [inline dev-docs/source/VSCode.rst]:
```suggestion
    }
```

File: dev-docs/source/VSCode.rst [hunk 4]
```diff
@@ -186,7 +190,36 @@ you will want to make your file look something a little like this:
...
+      "type": "cppdbg",
+      "request": "launch",
+      "program": "/usr/bin/python2.7", // Path to your used Python interpreter, here and below
+      "args": ["Path/To/Build/Directory/bin/workbench", "&&","gdb","/usr/bin/python2.7","$!"], // $! gets the process ID
+      "stopAtEntry": false,
+      "cwd": "Path/To/Build/Directory/bin", // this should point to bin inside the build directory
+      "environment": [],
+      "externalConsole": false,
+      "MIMode": "gdb",
+      "preLaunchTask": "Build Mantid",
```
Comment (Pasarus) [inline dev-docs/source/VSCode.rst]: Probably best to remove the preLaunchTask unless they actually want, some people won't.
```suggestion

```

**Review by Pasarus — CHANGES_REQUESTED** [review]
This is a great addition to the guide and is very very useful. A few small changes would be great though!

File: dev-docs/source/VSCode.rst (hunk 4 above)
Comment (DTasev) [reply dev-docs/source/VSCode.rst]: I've left it in so it matches the MantidPlot section, I think people can remove it from both if they don't want to do that

**Review by Pasarus — APPROVED** [review]

**Pull request merged**